components/edit_post/index.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+import {ActionCreatorsMapObject, bindActionCreators, Dispatch} from 'redux';
+
+import {addMessageIntoHistory} from 'mattermost-redux/actions/posts';
+import {Preferences, Permissions} from 'mattermost-redux/constants';
+import {getConfig} from 'mattermost-redux/selectors/entities/general';
+import {haveIChannelPermission} from 'mattermost-redux/selectors/entities/roles';
+import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
+import {getCurrentChannelId} from 'mattermost-redux/selectors/entities/channels';
+import {getCurrentUserId, isCurrentUserSystemAdmin} from 'mattermost-redux/selectors/entities/users';
+import {getBool} from 'mattermost-redux/selectors/entities/preferences';
+
+import {unsetEditingPost} from 'actions/post_actions';
+import {openModal} from 'actions/views/modals';
+import {scrollPostListToBottom} from 'actions/views/channel';
+import {editPost} from 'actions/views/posts';
+import {getEditingPost} from 'selectors/posts';
+import {GlobalState} from 'types/store';
+import Constants from 'utils/constants';
+
+import EditPost, {Actions} from './edit_post';
+
+function mapStateToProps(state: GlobalState) {
+    const config = getConfig(state);
+    const editingPost = getEditingPost(state);
+    const currentUserId = getCurrentUserId(state);
+    const channelId = editingPost?.post?.channel_id || getCurrentChannelId(state);
```

Comment:
yes, we do need it at the moment. The problem is that the selector `getEditingPost` cannot safely return a valid `editingPost` object. Otherwise we could just extract `post.channel_id` directly. Although adding a guard should not be too complicated. Let me see what I can do here.

Reply:
yep ... we can just assume that editingPost is always there when we mount the `EditPost` component, so there is no need for the fallback.